test/suite/test_cursor_bound02.py

```diff
@@ -0,0 +1,132 @@
+        else:
+            return tuple(tuple_key)
+
+    def test_bound_api(self):
+        uri = self.uri + self.file_name
+        create_params = 'value_format=S,key_format={}'.format(self.key_format)
+        self.session.create(uri, create_params)
+        cursor = self.session.open_cursor(uri)
+
+        # Test bound API: Basic usage.
+        cursor.set_key(self.gen_key(40))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+        cursor.set_key(self.gen_key(90))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: Upper bound < lower bound.
+        cursor.set_key(self.gen_key(30))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=upper"), '/Invalid argument/')
+
+        # Test bound API: Lower bound > upper bound.
+        cursor.set_key(self.gen_key(99))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=lower"), '/Invalid argument/')
+
+        # Test bound API: Test resetting lower bound to 20, which would succeed setting upper
+        # bound to 30
+        cursor.set_key(self.gen_key(20))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+        cursor.set_key(self.gen_key(30))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: Test that clearing the lower bound works.
+        cursor.set_key(self.gen_key(10))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=upper"), '/Invalid argument/')
+        self.assertEqual(cursor.bound("action=clear,bound=lower"), 0)
+        cursor.set_key(self.gen_key(10))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: Test that clearing the upper bound works. 
+        cursor.set_key(self.gen_key(20))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=lower"), '/Invalid argument/')
+        self.assertEqual(cursor.bound("action=clear,bound=upper"), 0)
+        cursor.set_key(self.gen_key(20))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+    
+        # Test bound API: Test that clearing both of the bounds works. 
+        cursor.set_key(self.gen_key(50))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+        self.assertEqual(cursor.bound("action=clear"), 0)
+        cursor.set_key(self.gen_key(90))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+        cursor.set_key(self.gen_key(99))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: No key set.
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=lower"), '/Invalid argument/')
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=upper"), '/Invalid argument/')
+        
+        # Test bound API: Test that reset also clears both of the bounds. 
+        cursor.reset()
```

Comment:
What happens when you set a new upper bound thats < the previous upper bound, but you don't do a reset. Is that expected to be valid behaviour? If so, does this actually test that reset() works as you'd be expected to be able to do this anyway.

Reply:
>What happens when you set a new upper bound thats < the previous upper bound, but you don't do a reset. Is that expected to be valid behaviour? 
Yep that should be total valid behaviour
>If so, does this actually test that reset() works as you'd be expected to be able to do this anyway.
In this case, we are checking if the lower bounds were cleared. Previously before the clearing the lower bounds were 90, thus placing upper bounds at 80 should return EINVAL, if the lower bound wasn't cleared.

Reply:
I misread this, this makes sense to me.